src/libraries/Microsoft.PowerFx.Json/Public/Types/UserProvidedTypeDefinitions/FormulaTypeSchema.cs

```diff
@@ -54,18 +55,28 @@ public FormulaType ToFormulaType(DefinedTypeSymbolTable definedTypeSymbols)
                 return actualType;
             }
 
-            if (typeName != SchemaTypeName.RecordTypeName.Name)
+            var logicalNameToRecordType = settings.LogicalNameToRecordType;
+            if (typeName == SchemaTypeName.ExpandableTableTypeName.Name && logicalNameToRecordType != null)
             {
-                return FormulaType.BindingError;
+                return logicalNameToRecordType.Invoke(this.Description).ToTable();
             }
-
-            if (Fields == null || !Fields.Any())
+            else if (typeName == SchemaTypeName.ExpandableRecordTypeName.Name && logicalNameToRecordType != null)
+            {
+                return logicalNameToRecordType.Invoke(this.Description);
+            }
+            else if (typeName == SchemaTypeName.RecordTypeName.Name)
```

Comment:
IMO, we shouldn't be using "ExpandTable" + the description to handle serializing the types of related entities. 

Instead, consider adding those types to a definedTypeSymbolTable, and using the type names (which, in discussion with Greg when doing the Custom Page work, we determined would be the registered table names). 

i.e. 

a table *[Foo: E] where E is to the Accounts table and is tabular, should be serialized as 

```
{
    Type: {
         TypeName: Record
          IsTable: true
     }
      Fields: 
      {
             Foo:
             {
                 Type: {
                      TypeName: Accounts
                      IsTable: true
                   }
             }
      }
}
````

Reply:
Discussed offline.